Src/WindowsServer/WindowsServer.Shared/Implementation/AzureComputeMetadataHeartbeatPropertyProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    internal class AzureComputeMetadataHeartbeatPropertyProvider
+    {
+        internal const string HeartbeatPropertyPrefix = "azInst_"; // to ensure no collisions with base heartbeat properties
+
+        /// <summary>
+        /// Expected fields extracted from Azure IMS to add to the heartbeat properties. 
+        /// Set as internal for testing.
+        /// </summary>
+        internal readonly IReadOnlyCollection<string> ExpectedAzureImsFields = new List<string>()
```

Comment:
I know we've been here before, but I recommend using a string array instead of a List. The reason being the List initializer syntax expands into a series of calls to `Add`. It will result in resizing the List as it goes. The string array initializer will set the size up-front. Just change `new List<string>()` to `new string[]`

Reply:
Done. Best not to ever-expand in this case for sure.